nbconvert/exporters/exporter.py

```diff
@@ -308,4 +308,5 @@ def _preprocess(self, nb, resources):
         #to each preprocessor
         for preprocessor in self._preprocessors:
             nbc, resc = preprocessor(nbc, resc)
+            nbformat.validate(nbc, relax_add_props=True)
```

Comment:
That is surprisingly non-trivial to do given the way that `NotebookValidationError` is written.